api/src/services/setup/upgrade-log.js

```diff
@@ -0,0 +1,231 @@
+    logger.info('Last upgrade log is already final');
+    return;
+  }
+
+  currentUpgradeLogId = upgradeLog._id;
+
+  return upgradeLog;
+};
+
+/**
+ * Returns the user and the _id of the current upgrqde log
+ * @return {Promise<{upgrade_log_id: string, user: string}>}
+ */
+const getDeployInfo = async () => {
+  const log = await module.exports.get() || {};
+  return {
+    user: log.user,
+    upgrade_log_id: log._id,
+  };
+};
+
+/**
+ * Updates the upgrade log with a new state
+ * @param {UpgradeLog} upgradeLog
+ * @param {string} state
+ * @param {number} date
+ */
+const pushState = (upgradeLog, state, date = new Date().getTime()) => {
+  upgradeLog.state_history = upgradeLog.state_history || [];
+  upgradeLog.state = state;
+  upgradeLog.updated_date = date;
+  upgradeLog.state_history.push({ state, date });
+};
+
+/**
+ * Creates, saves and returns contents of a new upgrade log file
+ * @param {string} toVersion
+ * @param {string} fromVersion
+ * @param {string} username
+ * @return {Promise<UpgradeLog>}
+ */
+const createUpgradeLog = async (action, toVersion = '', fromVersion = '', username = '') => {
+  logger.info(`Staging ${toVersion}`);
+  const startDate = new Date().getTime();
+
+  /**
+   * @type UpgradeLog
+   */
+  const upgradeLog = {
+    _id: getUpgradeLogId(toVersion, startDate),
+    user: username,
+    action,
+    from_version: fromVersion,
+    to_version: toVersion,
+    start_date: startDate,
+  };
+  pushState(upgradeLog, UPGRADE_LOG_STATES.INITIATED, startDate);
+
+  await db.medicLogs.put(upgradeLog);
+  currentUpgradeLogId = upgradeLog._id;
+  return upgradeLog;
+};
+
+/**
+ * Updates the current upgrade log to set the new state
+ * Updates are skipped if the current log is already in a final state.
+ * If the new state is final, the log is no longer stored as "current"
+ * @param {string} state
+ * @returns {Promise}
+ */
+const update = async (state) => {
+  const upgradeLog = await module.exports.get();
+  if (!upgradeLog) {
+    logger.info('Upgrade log tracking file was not found.');
+    return;
+  }
+
+  if (isFinalState(upgradeLog.state)) {
+    return;
+  }
+
+  pushState(upgradeLog, state);
+  await db.medicLogs.put(upgradeLog);
+
+  if (isFinalState(upgradeLog.state) && currentUpgradeLogId === upgradeLog._id) {
+    currentUpgradeLogId = undefined;
+  }
+
+  return upgradeLog;
+};
+
+const setStaged = async () => {
+  await update(UPGRADE_LOG_STATES.STAGED);
+  logger.info('Staging complete');
+};
+
+const setIndexing = async () => {
+  logger.info('Indexing staged views');
+  await update(UPGRADE_LOG_STATES.INDEXING);
+};
+
+const setIndexed = async () => {
+  logger.info('Indexing views complete');
+  await update(UPGRADE_LOG_STATES.INDEXED);
+};
+
+const setCompleting = async () => {
+  logger.info('Completing install');
+  await update(UPGRADE_LOG_STATES.COMPLETING);
+};
+
+const setComplete = async () => {
+  await update(UPGRADE_LOG_STATES.COMPLETE);
+  logger.info('Install complete');
+};
```

Comment:
It looks like these functions are not called anywhere - are they just left over from the `finalize` rename?

Reply:
They're going to get called in the "complete" endpoint of the API (where the services are restarted). 
That part is not developed yet.